pennylane/transforms/condition.py

```diff
@@ -60,11 +61,32 @@ def __init__(
 
 
 def cond(condition, true_fn, false_fn=None):
-    """Condition a quantum operation on the results of mid-circuit qubit measurements.
+    """Quantum-compatible if-else conditionals --- condition quantum operations
+    on parameters such as the results of mid-circuit qubit measurements.
 
-    Support for using :func:`~.cond` is device-dependent. If a device doesn't
-    support mid-circuit measurements natively, then the QNode will apply the
-    :func:`defer_measurements` transform.
+    When used with the :func:`~.qjit` decorator with a hybrid
+    quantum-classical compiler, this function allows for general
+    if-else constructs. Both the ``true_fn`` and ``false_fn`` branches
+    will be captured by the compiler, with the executed branch determined
+    at runtime (similar to ``jax.lax.cond``). For more details, as well
+    as ``elseif`` support, please see :func:`catalyst.cond`.

```

Comment:
```suggestion
    at runtime (similar to ``jax.lax.cond``). For more details, as well
    as ``elif`` support, please see :func:`catalyst.cond`.
```

This is a good point, how would you do elif branches with this setup? I think we need to extend the signature no?

Reply:
Oops! Thanks for the suggestion! Also, for the scope of soft-integration, we didn't plan to update the signature of `cond` to accept any arbitrary number of `elif` branches. The main reason is encouraging users to use native Python `if-elif-else` (backed by AutoGraph) more and more instead of using `cond`.

Reply:
@dime10 Do you think that we have to update the docs to better address this or consider expanding the signature of `qml.cond`? cc: @josh146

Reply:
I am happy for `qml.cond` to be less functional than `catalyst.cond` (for now), as updating `qml.cond` will have consequences across demos/features/etc.

My assumption is that people will be trying out autograph instead, yes. Based on my limited experience @dime10, even though we say its experimental, it is just so much more accessible that I go straight to it these days first.

Reply:
What about adding an additional keyword argument for the `elif` functions, so `qml.cond(condition, true_fn, false_fn=None, elif_fns=())`? That shouldn't change any existing code.

Reply:
@dime10 how would you encode the `elif` conditions? Like the following?

```python
qml.cond(m == 1, true_fn, false_fn, elif_fns=((m==0, elif_fn1), (m==2, elif_fn2), ...))
```

I'm open to it, but I guess would still prefer to leave this for a followup PR after the integration work, so it can be reviewed separately 🤔

Reply:
cc @trbromley

Reply:
this would be pretty simple implementation, I realize.

Reply:
Sure we can leave it to a follow up 👍 

Either
`elif_fns=((m==0, elif_fn1), (m==2, elif_fn2), ...)`
or
`elif_fns=(elif_fn1, elif_fn2), elif_conditions=(m==0, m==2))`
would be fine I think and not too disruptive.

The implementation would indeed be trivial, just forwarding the arguments correctly to the Catalyst form :)